pkg/reconciler/revision/resources/deploy.go

```diff
@@ -228,6 +232,18 @@ func MakeDeployment(rev *v1.Revision,
 		return nil, fmt.Errorf("failed to create PodSpec: %w", err)
 	}
 
+	replicaCount := autoscalerConfig.InitialScale
+	ann, found := rev.ObjectMeta.Annotations[autoscaling.InitialScaleAnnotationKey]
+	if found {
+		initialScale, err := strconv.Atoi(ann)
+		if err != nil {
+			return nil, fmt.Errorf("failed to process initialScale annotation: %w", err)
+		}
+		if initialScale != 0 || autoscalerConfig.AllowZeroInitialScale {
+			replicaCount = int32(initialScale)
```

Comment:
should this already be validated, in general? I.e. can we just do `replicaCount = initialScale` ?

Reply:
Hmm the annotation is validated at the time of revision creation, but even if the cluster flag has changed, we will still preserve the replicaCount during reconciliation because the autoscaler will set it. I think we can skip the validation here.

Reply:
This only matters during creation, no?

Reply:
yes

Reply:
So I guess we can ignore the validation here.

Reply:
Sounds good. We also can benefit from the PA accessor on initialScale in #8353 so I'll hold this PR based on that change

Reply:
Actually we can't use the accessor because it's on PA 🤦‍♀️ I updated this to skip validation.